taichi/ir/snode.cpp

```diff
@@ -293,4 +295,10 @@ void SNode::set_index_offsets(std::vector<int> index_offsets_) {
   this->index_offsets = index_offsets_;
 }
 
+// TODO: rename to is_sparse?
+bool SNode::need_activation() const {
+  return type == SNodeType::pointer || type == SNodeType::hash ||
+         type == SNodeType::bitmasked || type == SNodeType::dynamic;
+}
```

Comment:
Not a new function - moved from `struct_llvm.cpp`. (Not sure why the definition was originally in the wrong file...)